securebanking-openbanking-uk-gateway-conversion-filter/src/main/java/com/forgerock/securebanking/uk/gateway/conversion/filter/IntentConverterFilter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.uk.gateway.conversion.filter;
+
+import com.forgerock.securebanking.openbanking.uk.common.api.meta.share.IntentType;
+import com.forgerock.securebanking.uk.gateway.conversion.factory.ConverterFactory;
+import org.forgerock.http.Filter;
+import org.forgerock.http.Handler;
+import org.forgerock.http.protocol.Request;
+import org.forgerock.http.protocol.Response;
+import org.forgerock.http.protocol.ResponseException;
+import org.forgerock.http.protocol.Status;
+import org.forgerock.openig.heap.GenericHeaplet;
+import org.forgerock.services.context.Context;
+import org.forgerock.util.promise.NeverThrowsException;
+import org.forgerock.util.promise.Promise;
+import org.forgerock.util.promise.PromiseImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+
+import static org.forgerock.http.protocol.Response.newResponsePromise;
+import static org.forgerock.json.JsonValueFunctions.enumConstant;
+
+/**
+ * Filter to convert IDM json intent objects to OB data model objects.
+ *
+ * This filter must have received {@code intentType} as required to identify the intent type {@link IntentType#toString()} to instance the converter,
+ * the {@code intentContent} as optional of string representation of a json intent to be converter to OB data model object,
+ * if 'intentContent' isn't provided the filter will get the intent content in string format from the entity request.
+ * Failures from the `Converter instance` is either a {@link RuntimeException} that will catch to build a {@link ResponseException}.
+ *
+ * Configuration options:
+ *
+ * <pre>
+ * {@code {
+ *      "type": "ConversionToOBObjectFilter",
+ *      "config": {
+ *         "intentType"     IntentType#toString()    [REQUIRED - (ItIdentifies the consent type by enum name (ACCOUNT_ACCESS_CONSENT, PAYMENT_INTERNATIONAL_CONSENT... ]
+ *         "intentContent"  string                   [OPTIONAL - String representation of json intent object to be converted to OB Object]
+ *      }
+ *  }
+ *  }
+ * </pre>
+ */
+public class IntentConverterFilter implements Filter {
+
+    private static final Logger logger = LoggerFactory.getLogger(IntentConverterFilter.class);
+
+    private final IntentType intentType;
+    private final String intentContent;
```

Comment:
What is the use case for passing the intentContent string as config? Won't we always want to take the intent from the http request/response body?

Reply:
No always we want to get the intent from response body, could exist a case that we want to convert only part of the IDM intent content, as example: `request.getEntity().getJson().<key>.<key>.toString()` to convert only part of the intent entity or the case that we have the intent content in a variable or property that we can inject in the Filter using the configuration.

Reply:
Deleted that logic, now the configuration will have a field to indicate where the filter needs to get the incoming payload.
```json
{
    "name": "IntentConverterFilter-accessAccountConsent"
     "type": "IntentConverterFilter",
    "config": {
        "intentType": "ACCOUNT_ACCESS_CONSENT",
        "payloadFrom":"REQUEST",
        "resultTo": ["RESPONSE"]
    }
}
```

Reply:
Configuration changed:
```json
{
  "name": "IntentConverterFilter-AccountAccessConsent",
  "type": "IntentConverterFilter",
  "config": {
    "messageType": "request",
    "entity": "#{request.entity.string}"
    "resultTo": [
      "response"
    ]
  }
}
```
## Configuration

| field         | description                                                                                   | required | default         |
|---------------|-----------------------------------------------------------------------------------------------|----------|-----------------|
| `messageType` | The type of message for which to convert the entity                                           | Yes      | N/A             |
| `entity`      | Expression<String> to be replaced for a jsonPayload content to be converted in string format. | No       | [not replaced.] |
| `resultTo`    | Indicates where will set the conversion result                                                | No       | `REQUEST`       |